Add: Azure Active Directory SSO (#1910)
getsentry/sentry-docs: gatsby/src/docs/accounts/sso/azure-sso.mdx

---
title: 'Azure Active Directory SSO'
sidebar_order: 1
---

## Installation

1. Log in to the [Azure portal](https://portal.azure.com/).

2. (If you don't require a permission group for Sentry, skip to Step 3.) Under Manage Azure Active Directory (the shield picture), click "View", which takes you to the Organization Overview page. Now, proceed to Step 3.

3. On the left hand navigation, click "Groups". Create a new group, add an owner, and members to the group. Set "Group type" to Office 365. For more details about group creation, see the [Azure docs](https://docs.microsoft.com/en-us/azure/active-directory/fundamentals/active-directory-groups-create-azure-portal).

> 4. Return to the Overview page. In the search bar, search for "Enterprise Applications" and navigate to "Enterprise Applications." Then, select "Non-gallery application" and create a non-gallery application for Sentry. Once created, you'll be directed to "Sentry - Overview.

Author: @vudngo I rewrote this step. Let me know if we can improve it. 😄

Reviewer: this is perfect.

> 4. Return to the Overview page. In the search bar, search for "Enterprise Applications" and navigate to "Enterprise Applications." Then, select "Non-gallery application" and create a non-gallery application for Sentry. Once created, you'll be directed to "Sentry - Overview.

Author: ```suggestion
4. Return to the Overview page. In the search bar, search for "Enterprise Applications" and navigate to "Enterprise Applications." Click "+ New application". Then, select "Non-gallery application" and create a non-gallery application for Sentry. Once created, you'll be directed to "Sentry - Overview.
```

    ![Sentry Overview](azure-sentry-overview.png)

> 5. Click on `1. Assign users and groups`, and assign yourself and the group you've created to the Sentry app.

Author: ```suggestion
5. Click on `1. Assign users and groups`, with "+ Add user" add yourself and the group you've created to the Sentry app. Click "Assign".
```

> 6. Click on `2. Setup Single Sign-on`.

Author: ```suggestion
6. Navigate back to "Overview" and click on `2. Setup Single Sign-on`.
```

> 7. For Section (1) labeled as `Basic SAML2 Configuration`, enter the following data in each line.    

Author: ```suggestion
7. For Section (1) labeled as `Basic SAML Configuration`, enter the following data in each line and save your changes.    
```

    - Identifier (entity ID): https://sentry.io/saml/metadata/sentry/

    - Reply URL (Assertion Consumer Service URL): https://sentry.io/saml/acs/sentry/ 

    - Sign on URL: https://sentry.io/auth/login/sentry/

    - Relay State: https://sentry.io/

    - Logout URL: https://sentry.io/saml/sls/sentry/

    ![SAML Configuration](azure-basic-saml-configuration.png)

8. In Section (3) labeled as `SAML Signing Certificate`, copy the `App Federation Metadata URL`.

>     ![SAML Signing Certificate](azure-saml-signing-certificate.png)

Reviewer 2: It'll be a good idea to blank out the email here to avoid confusion.

9. Navigate to your **Org Settings > Auth > Azure** (or go to `https://sentry.io/settings/<YOUR-ORGANIZATION-SLUG>/auth/`) and click on "Configure" for Active Directory.

10. Paste the App Federation Metadata URL and click "Get metadata".

11. In the following page, enter these to map the values from AzureAD to Sentry.

    - IdP User ID: http://schemas.xmlsoap.org/ws/2005/05/identity/claims/name

    - User Email: http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress

    - First Name: http://schemas.xmlsoap.org/ws/2005/05/identity/claims/givenname

>     - Last Name: http://schemas.xmlsoap.org/ws/2005/05/identity/claims/surname

Reviewer 2: Would it be okay to wrap these in a code block so the "addresses" doesn't get converted into a link? 

They aren't hyperlinks and doesn't go anywhere valid.

---
title: 'Azure Active Directory SSO'
sidebar_order: 1
---

## Installation

1. Log in to the [Azure portal](https://portal.azure.com/).

> 2. (If you don't require a permission group for Sentry, skip to Step 3.) Under 

Reviewer 2: ```suggestion
2. (If you don't require a permission group for Sentry, skip to Step 4.) Under 
```

Manage Azure Active Directory (the shield picture), click "View", which takes
you to the Organization Overview page. Now, proceed to Step 3.

> 3. On the left hand navigation, click "Groups". Create a new group, add an owner, and

Author: Sure! Would we write something like:

3. In the search bar, search for "Groups" and navigate to "Groups." From there, create a new group, add an owner, and members to the group. Set "Group type" to Office 365. For more details about group creation, see the [Azure docs](https://docs.microsoft.com/en-us/azure/active-directory/fundamentals/active-directory-groups-create-azure-portal).

> 3. On the left hand navigation, click "Groups". Create a new group, add an owner, and

Reviewer: that's perfect.

members to the group. Set "Group type" to Office 365. For more details about group creation, see the
[Azure docs](https://docs.microsoft.com/en-us/azure/active-directory/fundamentals/active-directory-groups-create-azure-portal).

4. Return to the Overview page. On the left hand navigation, click "Enterprise Applications".
Then, select "Non-gallery application" and create a non-gallery application for Sentry. Once created, you'll
be directed to "Sentry - Overview."

5. Click on `1. Assign users and groups`, and assign yourself and the group you've created 
to the Sentry app.

6. Click on `2. Setup Single Sign-on`

7. For Section (1) labeled as `Basic SAML2 Configuration`, enter the following data in each line.    

>     - Identifier (entity ID): https://sentry.io/saml/metadata/sentry/

Author: I'm gonna defer to @leedongwei for this one. 😄

>     - Identifier (entity ID): https://sentry.io/saml/metadata/sentry/

Reviewer 2: Good catch @vudngo, it should be their own `org-slug`. 

See: https://docs.sentry.io/accounts/sso/saml2/#register-sentry-with-idp

    - Reply URL (Assertion Consumer Service URL): https://sentry.io/saml/acs/sentry/ 

    - Sign on URL: https://sentry.io/auth/login/sentry/

    - Relay State: https://sentry.io/

>     - Logout URL: https://sentry.io/saml/sls/sentry/

Reviewer 2: Same thing: `https://sentry.io/.../.../YOUR-ORG-SLUG/`

8. In Section (3) labeled as `SAML Signing Certificate`, copy the `App Federation Metadata URL`

> 9. Go to `https://sentry.io/settings/<YOUR-ORGANIZATION-SLUG>/auth/` and click on "Configure" 

Author: We could provide both options. What navigational directions would you write here?

> 9. Go to `https://sentry.io/settings/<YOUR-ORGANIZATION-SLUG>/auth/` and click on "Configure" 

Reviewer: I would say something like, "go to your Org Settings -> Auth -> Azure".
